app/components/RerunService.js

```diff
@@ -0,0 +1,122 @@
+"use client";
+
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import Button from "./Button";
+import ConfirmModal from "./ConfirmModal";
+
+// Sets a message in localStorage to be displayed as an alert.
+export function setLocalStorage(message, color, icon) {
+  localStorage.setItem(
+    "zodiacMessage",
+    JSON.stringify({
+      message: message,
+      color: color,
+      icon: icon,
+    }),
+  );
+}
+
+// Sends a post request to restart a service for a specific package.
+export function rerunService(service, package_id, router) {
+  const postData = {
+    service: service,
+    package_id: package_id,
+  };
+  fetch(`${process.env.NEXT_PUBLIC_API_BASE_URL}/restart-service/`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(postData),
+  })
+    // If the request is successful, sets a success message in localStorage and redirects to the package page.
+    .then((resp) => {
+      if (resp.ok) {
+        setLocalStorage(
+          "The request to rerun the service has been received. It may take a few minutes for a notification from this service to appear. Please refresh the page for status updates",
+          "blue",
+          "check_circle_outline",
+        );
+        router.push(`/packages/${package_id}`);
+        // If the request fails, sets an error message in localStorage and reloads the page.
+      } else {
+        resp.json().then((data) => {
+          localStorage.setItem(
+            "zodiacMessage",
+            JSON.stringify({
+              message: data,
+              color: "orange",
+              icon: "error_outline",
+            }),
+          );
+          window.location.reload();
+        });
+      }
+    })
+    // If there is a network error, sets an error message in localStorage and reloads the page.
+    .catch((error) => {
+      localStorage.setItem(
+        "zodiacMessage",
+        JSON.stringify({
+          message: String(error),
+          color: "orange",
+          icon: "error_outline",
+        }),
+      );
+      window.location.reload();
+    });
```

Comment:
Can we create a common function like `function handleError(errorMessage) { ... }` to use in these so that we don't have to repeat the `localStorage.setItem( ` logic twice here?